### ISSUE TITLE
Cannot filter using accent Insensitive

### ISSUE THREAD

**Issue by sostenesapollo** [prisma/prisma#16211]
### Bug description

I've looked in all the places trying to find a way to use `{where: {field: {contains: 'João' } } ` and retrieve all the results without matching of `accents`, 

The result I expect:
i.e : `João` = `Joao`

I've tried to add a collation manually in the field that I'm searching, like this:
```
// Creating the collation
CREATE COLLATION ignore_accent (provider = icu, locale = 'und-u-ks-level1-kc-true', deterministic = false);

// Creating the table with the collation already applied in the field
CREATE TABLE "protocols" (
    "id" SERIAL NOT NULL,
    "title" TEXT NOT NULL COLLATE ignore_accent,

    CONSTRAINT "protocols_pkey" PRIMARY KEY ("id")
);
```

But It is showing me this error
```
ConnectorError(ConnectorError {
 user_facing_error: None, kind: QueryError(Error {
 kind: Db, cause: Some(DbError { severity: "ERROR", 
parsed_severity: Some(Error), code: SqlState(E0A000), 
message: "nondeterministic collations are not supported for ILIKE", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("like.c"), line: Some(199), routine: Some("Generic_Text_IC_like") 
}) }) })
```

### How to reproduce

.

### Expected behavior

.

### Prisma information

ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E0A000), message: "nondeterministic collations are not supported for ILIKE", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("like.c"), line: Some(199), routine: Some("Generic_Text_IC_like") }) }) })

### Environment & setup

.

### Prisma Version

.

**Comment by Jolg42** [comment]
Note:
This looks like PostgreSQL >= 12

https://www.postgresql.org/docs/current/collation.html#COLLATION-NONDETERMINISTIC
https://postgresql.verite.pro/blog/2019/10/14/nondeterministic-collations.html

@sostenesapollo 
Which version of PostgreSQL do you use?
Which Prisma version are you running? (You can run `npx prisma -v`)

**Comment by sostenesapollo** [comment]
This version @Jolg42  `"@prisma/client": "^3.15.2"`

**Comment by NikitaMasl** [comment]
@sostenesapollo, did you find a solution?

**Comment by NikitaMasl** [comment]
@sostenesapollo, yeah, It was my first thought.
But then I changed deterministic to true during it creation in migration and everything start working. Searching with LIKE and ILIKE and sorting stay case insensitive

**Comment by sostenesapollo** [comment]
@NikitaMasl  yeah, I've tried everything you can imagine, and it seems to be a prisma limitation

**Comment by sostenesapollo** [comment]
I think if you switch to mysql you'll be able to run with your custom collation or something like this

**Comment by sostenesapollo** [comment]
@NikitaMasl I found a workarround using chatGTP lol

**Comment by sostenesapollo** [comment]
<img width="855" alt="image" src="https://user-images.githubusercontent.com/35838667/209451035-ff63e7f0-1d58-4118-a6b4-89ae1ac59bf5.png">
This was his answer
And my implementation:

```
const name = 'Sóstenes Apollo';
const filterText = removeAccents(`%${name}%`).toLowerCase();
console.log('Filter text', filterText);

this.prisma.$queryRaw`select 
    id, name 
  from users 
    where LOWER(unaccent(name)) LIKE ${filterText};`.then((users) =>
  console.log(users),
);
```

ps. note that the removeAccents method will return like: `Sostenes Apollo`, so I just pick it and make it lower to be like: `sostenes apollo`

**Comment by sostenesapollo** [comment]
`-- CREATE EXTENSION unaccent;`
Youl'd need to set this command if you're using postgres.

**Comment by gabrielew** [comment]
I created a migration to handle with this issue

```js
-- DropForeignKey
DROP EXTENSION IF EXISTS unaccent;

-- AddForeignKey
CREATE EXTENSION unaccent;
```

Unfortunately, this only works with **query raw**!

**Comment by martinratinaud** [comment]
Well this is really unfortunate, is there any ETA for this?

**Comment by matmkian** [comment]
Being able to filter by ignoring accent is key also in France :-) Search is not usable without supporting this extension :-(

**Comment by schieck0** [comment]
It's a shame, they don't deserve to be called developers

**Comment by martinratinaud** [comment]
> It's a shame, they don't deserve to be called developers

Well that's a bit harsh and not fair, remember it's open source software.
But I agree with @matmkian (I'm french too) and without this, it's difficult to consider prisma as properly useable.

**Comment by theorenck** [comment]
Hey everyone, any updates on this? Do we have an ETA yet?

**Comment by larafale** [comment]
need this so bad !
